pkg/segment/structs/evaluationstructs.go

```diff
@@ -181,20 +181,21 @@ type StringExpr struct {
 }
 
 type TextExpr struct {
-	IsTerminal  bool
-	Op          string //lower, ltrim, rtrim
-	Param       *StringExpr
-	StrToRemove string
-	Delimiter   *StringExpr
-	ValueList   []*StringExpr
-	StartIndex  *NumericExpr
-	EndIndex    *NumericExpr
-	LengthExpr  *NumericExpr
-	Val         *ValueExpr
-	Condition   *BoolExpr  // To filter out values that do not meet the criteria within a multivalue field
-	InferTypes  bool       // To specify that the mv_to_json_array function should attempt to infer JSON data types when it converts field values into array elements.
-	Cluster     *Cluster   // generates a cluster label
-	SPathExpr   *SPathExpr // To extract information from the structured data formats XML and JSON.
+	IsTerminal     bool
+	Op             string //lower, ltrim, rtrim
+	Param          *StringExpr
+	StrToRemove    string
+	Delimiter      *StringExpr
+	ValueList      []*StringExpr
+	StartIndex     *NumericExpr
+	EndIndex       *NumericExpr
+	LengthExpr     *NumericExpr
+	Val            *ValueExpr
+	Condition      *BoolExpr       // To filter out values that do not meet the criteria within a multivalue field
+	InferTypes     bool            // To specify that the mv_to_json_array function should attempt to infer JSON data types when it converts field values into array elements.
+	Cluster        *Cluster        // generates a cluster label
+	SPathExpr      *SPathExpr      // To extract information from the structured data formats XML and JSON.
+	EventCountExpr *EventCountExpr // To count the number of events in an index
```

Comment:
Does `EventCountExpr` belong in the `TextExpr` like you have it or in the `NumericExpr`? I forget if the type refers to the type that expression returns or the type that expression takes as inputs

Reply:
I think it doesn't belong to `TextExpr` or `NumericExpr` since we're not doing any text or numeric manipulation on the results. I created another Request type called `EventCountRequest` in `LetColumnsRequest`, similar to what we have for `FormatResults` in `LetColumnRequest`